### PR TITLE
Disable tor by group policy on Windows.

### PR DESCRIPTION
**Pull request by simonhong**
Introduce TorDisabled group policy for disabling tor feature.
also disable Tor.
This should be merged with https://github.com/brave/brave-browser/pull/6475.

Fix https://github.com/brave/brave-browser/issues/454
Fix https://github.com/brave/brave-browser/issues/6044

When `TorDisabled` policy is set to registry (`HKCU|HKLM\\Software\\Policies\\BraveSoftware\\Brave`),
 new menu item is visible in app menu like below.
Annotation 2019-10-15 180215

brave branding `brave://management` webui page that can be opened by above menu item.
Annotation 2019-10-15 193105

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- `yarn test brave_browser_tests --filter=*TorDisabled*`
- also see https://github.com/brave/brave-browser/issues/454

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [x] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by simonhong** [comment]
I think all entry point for tor window are disabled and hidden. Also, tor component updater is not registered when tor is disabled by gpo.
According to the spec(https://docs.google.com/document/d/1xlfgDbk95aPmLNOQWwqIQtpvYuqACGVUgOg0nv4opZw), one last thing I should do is delete tor binary if exists.
For this, I need help from tor expert. (I don't know what tor binary is)
cc @bsclifton @darkdh @yrliou

File: patches/components-policy-tools-generate_policy_source.py.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/components/policy/tools/generate_policy_source.py b/components/policy/tools/generate_policy_source.py
+index 28db4dd238ad33e4a250d93c1ef90678e5c31ef8..135515f639a3e4744c8f1efd7503cb802135bcc5 100755
+--- a/components/policy/tools/generate_policy_source.py
++++ b/components/policy/tools/generate_policy_source.py
+@@ -24,7 +24,7 @@ import types
+ from xml.sax.saxutils import escape as xml_escape
+ 
+ CHROME_POLICY_KEY = 'SOFTWARE\\\\Policies\\\\Google\\\\Chrome'
+-CHROMIUM_POLICY_KEY = 'SOFTWARE\\\\Policies\\\\Chromium'
++CHROMIUM_POLICY_KEY = 'SOFTWARE\\\\Policies\\\\Brave'
```
Comment (simonhong) [inline patches/components-policy-tools-generate_policy_source.py.patch]: Or this should be `SOFTWARE\\\\Policies\\\\BraveSoftware\\\\Brave` ?

Comment (simonhong) [reply patches/components-policy-tools-generate_policy_source.py.patch]: Changed to use `SOFTWARE\\\\Policies\\\\BraveSoftware\\\\Brave` because updater policy is set to `SOFTWARE\\\\Policies\\\\BraveSoftware\\\\Update`.

**Comment by simonhong** [comment]
This PR is ready to review.

File: common/pref_names.h [hunk 2]
```diff
@@ -59,6 +59,7 @@ extern const char kBraveWalletAES256GCMSivNonce[];
 extern const char kBraveWalletEncryptedSeed[];
 extern const char kBraveWalletEnabled[];
 extern const char kAlwaysShowBookmarkBarOnNTP[];
+extern const char kTorDisabled[];
```
[the remark this answers is not in the document]
Comment (simonhong) [reply common/pref_names.h]: IIUC, `brave/common/tor/pref_names.h` includes prefs that used in internal tor.
I think `kTorDisabled` is switch for on/off the tor feature itself.
So, I think adding it here is more appropriate. WDYT?

[the remark this answers is not in the document]
Comment (simonhong) [reply common/pref_names.h]: @darkdh  Hmm, after some thought, I think gathering tor related prefs into one place seems fine.
and I think local state seems more better place for this new prefs.
So, I tried to add as you suggested. but there are some issues.
`BraveExtensionManagement` is created before `TorProfileService:: RegisterLocalStatePrefs()` is called.
Because of this, I can't check `kTorDisabled` in `BraveExtensionManagement::RegisterBraveExtensions` or `CleanupBraveExtensions`.
I think we need some refactoring. WDYT?

[the remark this answers is not in the document]
Comment (darkdh) [reply common/pref_names.h]: Agree with LocalStatePrefs but I'm sure that `brave::RegisterLocalStatePrefs()` is called before `BraveExtensionManagement` being instantiated

[the remark this answers is not in the document]
Comment (simonhong) [reply common/pref_names.h]: Moved `kTorProfile` prefs to local state. Thanks!

**Comment by simonhong** [comment]
Kindly ping to reviewers :)

File: browser/policy/brave_policy_browsertest.cc [hunk 3]
```diff
@@ -83,7 +88,7 @@ IN_PROC_BROWSER_TEST_F(TorEnabledPolicyBrowserTest, TorDisabledPrefValueTest) {
 // W/o TorDisabled group policy, kTorDisabled pref value should be false.
 IN_PROC_BROWSER_TEST_F(NoTorPolicyBrowserTest,
                        DefaultTorDisabledPrefValueTest) {
-  EXPECT_FALSE(browser()->profile()->GetPrefs()->GetBoolean(kTorDisabled));
+  EXPECT_FALSE(tor::TorProfileService::IsTorDisabled());
```
Comment (darkdh) [inline browser/policy/brave_policy_browsertest.cc]: This test case also needs `BUILDFLAG(ENABLE_TOR)`

File: browser/ui/brave_browser_command_controller.cc [hunk 4]
```diff
@@ -119,8 +123,7 @@ void BraveBrowserCommandController::UpdateCommandForBraveAdblock() {
 }
 
 void BraveBrowserCommandController::UpdateCommandForTor() {
-  const bool isTorEnabled =
-      !browser_->profile()->GetPrefs()->GetBoolean(kTorDisabled);
+  const bool isTorEnabled = !tor::TorProfileService::IsTorDisabled();
```
Comment (darkdh) [inline browser/ui/brave_browser_command_controller.cc]:
```cpp
#if BUILDFLAG(ENABLE_TOR)
const bool isTorEnabled = !tor::TorProfileService::IsTorDisabled();
#else
const bool isTorEnabled = false
#endif
```

File: browser/policy/brave_policy_browsertest.cc (hunk 3 above)
Comment (simonhong) [reply browser/policy/brave_policy_browsertest.cc]: Done.

File: browser/ui/brave_browser_command_controller.cc (hunk 4 above)
Comment (simonhong) [reply browser/ui/brave_browser_command_controller.cc]: Done.

**Comment by bsclifton** [comment]
Added test plan under https://github.com/brave/brave-browser/issues/454 and corrected the milestone/labels for the issues 👍

File: chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc [hunk 5]
```diff
@@ -78,9 +85,11 @@ void BraveRenderViewContextMenu::AddSpellCheckServiceItem(
 void BraveRenderViewContextMenu::InitMenu() {
   RenderViewContextMenu_Chromium::InitMenu();
 
+#if BUILDFLAG(ENABLE_TOR)
+  const bool isTorEnabled = !tor::TorProfileService::IsTorDisabled();
```
Comment (bridiver) [inline chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc]: c++ vars use underscores, not camelcase

File: browser/extensions/brave_extension_management.cc [hunk 6]
```diff
@@ -43,11 +54,28 @@ BraveExtensionManagement::~BraveExtensionManagement() {
 
 void BraveExtensionManagement::RegisterBraveExtensions() {
 #if BUILDFLAG(ENABLE_TOR)
+  const bool isTorEnabled = !tor::TorProfileService::IsTorDisabled();
```
Comment (bridiver) [inline browser/extensions/brave_extension_management.cc]: nit - this var is not necessary

File: chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc (hunk 5 above)
Comment (bridiver) [reply chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc]: also this var seems unnecessary

File: browser/ui/brave_browser_command_controller.cc [hunk 7]
```diff
@@ -117,8 +123,14 @@ void BraveBrowserCommandController::UpdateCommandForBraveAdblock() {
...
 void BraveBrowserCommandController::UpdateCommandForTor() {
-  UpdateCommandEnabled(IDC_NEW_TOR_CONNECTION_FOR_SITE, true);
-  UpdateCommandEnabled(IDC_NEW_OFFTHERECORD_WINDOW_TOR, true);
+  const bool isTorEnabled =
+#if BUILDFLAG(ENABLE_TOR)
+      !tor::TorProfileService::IsTorDisabled();
+#else
+      false;
+#endif
+  UpdateCommandEnabled(IDC_NEW_TOR_CONNECTION_FOR_SITE, isTorEnabled);
```
Comment (bridiver) [inline browser/ui/brave_browser_command_controller.cc]: why would we call these at all if `BUILDFLAG(ENABLE_TOR)` is false? I think the original code should have been wrapped in `#if BUILDFLAG(ENABLE_TOR)` and then `true` can be replaced by `!tor::TorProfileService::IsTorDisabled()`

File: browser/ui/toolbar/brave_app_menu_model_browsertest.cc [hunk 8]
```diff
@@ -8,17 +8,24 @@
 #include <memory>
 
 #include "brave/app/brave_command_ids.h"
+#include "brave/browser/tor/buildflags.h"
```
Comment (bridiver) [inline browser/ui/toolbar/brave_app_menu_model_browsertest.cc]: is there a dep for this?

File: browser/ui/toolbar/brave_app_menu_model_browsertest.cc [hunk 9]
```diff
@@ -8,17 +8,24 @@
...
+#include "chrome/browser/chrome_browser_main.h"
+#include "chrome/browser/chrome_browser_main_extra_parts.h"
 #include "chrome/browser/chrome_notification_types.h"
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/profiles/profile_window.h"
 #include "chrome/browser/ui/browser_list.h"
 #include "chrome/browser/ui/views/frame/browser_view.h"
 #include "chrome/browser/ui/views/toolbar/toolbar_view.h"
 #include "chrome/test/base/in_process_browser_test.h"
+#include "components/prefs/pref_service.h"
```
Comment (bridiver) [inline browser/ui/toolbar/brave_app_menu_model_browsertest.cc]: is there a dep for this?

File: browser/ui/views/profiles/brave_profile_menu_view_helper.cc [hunk 10]
```diff
@@ -8,14 +8,17 @@
 #include "brave/browser/brave_browser_process_impl.h"
 #include "brave/browser/profiles/profile_util.h"
 #include "brave/browser/tor/buildflags.h"
+#include "brave/common/pref_names.h"
 #include "brave/grit/brave_generated_resources.h"
 #include "chrome/app/vector_icons/vector_icons.h"
+#include "components/prefs/pref_service.h"
```
Comment (bridiver) [inline browser/ui/views/profiles/brave_profile_menu_view_helper.cc]: why was this added?

File: browser/ui/views/profiles/brave_profile_menu_view_helper.cc [hunk 11]
```diff
@@ -8,14 +8,17 @@
 #include "brave/browser/brave_browser_process_impl.h"
 #include "brave/browser/profiles/profile_util.h"
 #include "brave/browser/tor/buildflags.h"
+#include "brave/common/pref_names.h"
```
Comment (bridiver) [inline browser/ui/views/profiles/brave_profile_menu_view_helper.cc]: why was this added?

File: chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc [hunk 12]
```diff
@@ -0,0 +1,52 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/stl_util.h"
+#include "brave/browser/tor/buildflags.h"
```
Comment (bridiver) [inline chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc]: is there a dep for this in the browser target we add to chrome/browser?

File: patches/components-policy-resources-policy_templates.json.patch [hunk 13]
```diff
@@ -0,0 +1,34 @@
+diff --git a/components/policy/resources/policy_templates.json b/components/policy/resources/policy_templates.json
+index 1aefb537e31087d645307d6bac4ef3448cda3e40..33a3a603447954f7ff647b65ce5daa3c5eb11309 100644
+--- a/components/policy/resources/policy_templates.json
++++ b/components/policy/resources/policy_templates.json
+@@ -17623,6 +17623,21 @@
+ 
+       See https://www.chromestatus.com/feature/4664843055398912 .''',
+     },
++    {
```
Comment (bridiver) [inline patches/components-policy-resources-policy_templates.json.patch]: we need to find a way to do this, or at the very least it needs to be compressed down to one line

File: browser/extensions/brave_extension_management.cc [hunk 14]
```diff
@@ -43,11 +54,28 @@ BraveExtensionManagement::~BraveExtensionManagement() {
...
+
+void BraveExtensionManagement::CleanupBraveExtensions() {
+#if BUILDFLAG(ENABLE_TOR)
+  // Delete tor binaries if tor is disabled by gpo.
+  if (tor::TorProfileService::IsTorDisabled()) {
+    ProfileManager* profile_manager = g_browser_process->profile_manager();
+    base::FilePath tor_component_dir =
+        profile_manager->user_data_dir().AppendASCII(
+            extensions::kTorClientComponentId);
+    g_brave_browser_process->tor_client_updater()->GetTaskRunner()->PostTask(
```
Comment (bridiver) [inline browser/extensions/brave_extension_management.cc]: The delete should be encapsulated in the tor_client_updater - `tor_client_updater()->Cleanup()`

File: patches/components-policy-resources-policy_templates.json.patch (hunk 13 above)
Comment (bridiver) [reply patches/components-policy-resources-policy_templates.json.patch]: I did see some comments that this is not a pure json file and supports some additional features so maybe we can utilize those or maybe parse this template, make changes to it and output again? Or maybe we can use a python include and insert directly into the parsed data?

Comment (bridiver) [reply patches/components-policy-resources-policy_templates.json.patch]: for instance maybe in generate_policy_source we can merge the original file with one we add here `template_file_contents = _LoadJSONFile(template_file_name)`?

File: patches/components-policy-resources-policy_templates.json.patch [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
++      'type': 'main',
++      'schema': { 'type': 'boolean' },
++      'supported_on': ['chrome.win:78-'],
++      'features': {
++        'dynamic_refresh': False,
++        'per_profile': False
++      },
++      'example_value': True,
++      'id': 605,
++      'caption': '''Allows to disable tor feature.''',
```
Comment (bridiver) [inline patches/components-policy-resources-policy_templates.json.patch]: nit: this isn't grammatically correct - `Disables the Tor feature`

File: chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc (hunk 5 above)
Comment (simonhong) [reply chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc]: Done.

File: browser/extensions/brave_extension_management.cc (hunk 6 above)
Comment (simonhong) [reply browser/extensions/brave_extension_management.cc]: Done.

File: browser/ui/brave_browser_command_controller.cc (hunk 7 above)
Comment (simonhong) [reply browser/ui/brave_browser_command_controller.cc]: Done and wrapped whole this method in BUILDFLAG.

File: browser/ui/toolbar/brave_app_menu_model_browsertest.cc (hunk 8 above)
Comment (simonhong) [reply browser/ui/toolbar/brave_app_menu_model_browsertest.cc]: Yes, it has like below.
```
simon-home:src simonhong$ gn path ./out/Debug brave/test:brave_browser_tests brave/browser/tor:buildflags
//brave/test:brave_browser_tests --[private]-->
//chrome/browser/ui:ui --[private]-->
//brave/browser/ui:ui --[private]-->
//brave/browser/tor:tor --[public]-->
//brave/browser/tor:buildflags
```

File: browser/ui/toolbar/brave_app_menu_model_browsertest.cc (hunk 9 above)
Comment (simonhong) [reply browser/ui/toolbar/brave_app_menu_model_browsertest.cc]: Yes
```
simon-home:src simonhong$ gn path ./out/Debug brave/test:brave_browser_tests components/prefs
//brave/test:brave_browser_tests --[private]-->
//chrome/browser/ui:ui --[public]-->
//components/dom_distiller/core:core --[private]-->
//components/prefs:prefs
```

File: browser/ui/views/profiles/brave_profile_menu_view_helper.cc (hunk 10 above)
Comment (simonhong) [reply browser/ui/views/profiles/brave_profile_menu_view_helper.cc]: Deleted. It was used directly at first but replaced with `tor::TorProfileService::IsTorDisabled()`

File: browser/ui/views/profiles/brave_profile_menu_view_helper.cc (hunk 11 above)
Comment (simonhong) [reply browser/ui/views/profiles/brave_profile_menu_view_helper.cc]: Ditto. deleted.

File: chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc (hunk 12 above)
Comment (simonhong) [reply chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc]: Yes.
```
simon-home:src simonhong$ gn path ./out/Debug/ chrome/browser  brave/browser/tor:buildflags
//chrome/browser:browser --[private]-->
//brave/browser:browser_process --[private]-->
//brave/browser/tor:tor --[public]-->
//brave/browser/tor:buildflags
```

File: patches/components-policy-resources-policy_templates.json.patch (hunk 13 above)
Comment (simonhong) [reply patches/components-policy-resources-policy_templates.json.patch]: Compressed to one line.

File: browser/extensions/brave_extension_management.cc (hunk 14 above)
Comment (simonhong) [reply browser/extensions/brave_extension_management.cc]: Refactored. Done.

File: patches/components-policy-resources-policy_templates.json.patch (hunk 15 above)
Comment (simonhong) [reply patches/components-policy-resources-policy_templates.json.patch]: Fixed.

**Comment by simonhong** [comment]
@bridiver Addressed all your comments except last comment regarding json template modifying.
I'm looking at it.

File: patches/components-policy-resources-policy_templates.json.patch (hunk 13 above)
Comment (simonhong) [reply patches/components-policy-resources-policy_templates.json.patch]: @bridiver `policy_source_helper.py` is introduced and it adds our policies. PTAL again.

File: patches/components-policy-tools-generate_policy_source.py.patch [hunk 16]
```diff
@@ -0,0 +1,24 @@
+diff --git a/components/policy/tools/generate_policy_source.py b/components/policy/tools/generate_policy_source.py
+index 28db4dd238ad33e4a250d93c1ef90678e5c31ef8..75ce4477f67a9c18de4202913b3b794c769fc800 100755
+--- a/components/policy/tools/generate_policy_source.py
++++ b/components/policy/tools/generate_policy_source.py
+@@ -22,9 +22,10 @@ import sys
+ import textwrap
+ import types
+ from xml.sax.saxutils import escape as xml_escape
++from policy_source_helper import AddBravePolicies
```
Comment (bridiver) [inline patches/components-policy-tools-generate_policy_source.py.patch]: is it possible to move this below `CHROMIUM_POLICY_KEY` and then import that as well to override it? I think it should work, but not 100% sure

File: script/policy_source_helper.py [hunk 17]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2019 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+def AddBravePolicies(template_file_contents):
```
Comment (bridiver) [inline script/policy_source_helper.py]: this is much better, thanks!

File: browser/ui/toolbar/brave_app_menu_model_browsertest.cc (hunk 9 above)
Comment (bridiver) [reply browser/ui/toolbar/brave_app_menu_model_browsertest.cc]: we should not depend on getting deps through a chain. They should always be added directly for the target

File: chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc (hunk 12 above)
Comment (bridiver) [reply chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc]: same as above, we should add a dep for `//brave/browser/tor:buildflags` directly to `//brave/browser:browser_process`
It actually seems quite weird that tor would have a dep on buildflags because tor should only be included if the buildflag is set

File: chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+#include "brave/common/pref_names.h"
+#include "brave/common/tor/pref_names.h"
+#include "chrome/browser/policy/configuration_policy_handler_list_factory.h"
+#include "components/policy/core/browser/configuration_policy_handler.h"
```
Comment (bridiver) [inline chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc]: same thing for these, we should have a dep in `//brave/browser:browser_process` for any includes we have

File: patches/components-policy-tools-generate_policy_source.py.patch (hunk 16 above)
Comment (simonhong) [reply patches/components-policy-tools-generate_policy_source.py.patch]: Oh, it works! 👍

File: chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc (hunk 18 above)
Comment (simonhong) [reply chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc]: Done.

File: chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc (hunk 12 above)
Comment (simonhong) [reply chromium_src/chrome/browser/policy/configuration_policy_handler_list_factory.cc]: Done.
```
//chrome/browser:browser --[private]-->
//brave/browser:browser_process --[private]-->
//brave/browser/tor:buildflags
```
And we added `tor` into two places. One is unconditioned deps list and the other is in `enable_tor`. So, `tor` in unconditioned deps list was removed.

File: browser/ui/toolbar/brave_app_menu_model_browsertest.cc (hunk 9 above)
Comment (simonhong) [reply browser/ui/toolbar/brave_app_menu_model_browsertest.cc]: Added.

**Comment by simonhong** [comment]
Kindly ping to @bridiver :)

File: browser/extensions/brave_extension_management.cc [hunk 19]
```diff
@@ -43,14 +43,18 @@ BraveExtensionManagement::~BraveExtensionManagement() {
 
 void BraveExtensionManagement::RegisterBraveExtensions() {
 #if BUILDFLAG(ENABLE_TOR)
-  const base::CommandLine& command_line =
-      *base::CommandLine::ForCurrentProcess();
-  if (!command_line.HasSwitch(switches::kDisableTorClientUpdaterExtension) &&
-      !profile_->AsTestingProfile())
+  if (!profile_->AsTestingProfile())
```
Comment (bridiver) [inline browser/extensions/brave_extension_management.cc]: I think you have some kind of rebase issue in here cc @darkdh

Comment (simonhong) [reply browser/extensions/brave_extension_management.cc]: @bridiver , I rebased this PR onto latest master yesterday.

Comment (darkdh) [reply browser/extensions/brave_extension_management.cc]: Why did you remove the switch check `kDisableTorClientUpdaterExtension`? Current master still has that switch.
The latest commit only added `AsTestingProfile` check

Comment (simonhong) [reply browser/extensions/brave_extension_management.cc]: Other conditions are moved to `BraveTorClientUpdater::Register()`. - https://github.com/brave/brave-core/pull/3692/commits/81bae87c04cc0572256df7fdb34e26155cfd757a#diff-5293ed891460b117107c7923e61926efR127
The intension of this refactoring is moving tor update specific logic to `BraveTorClientUpdater::Register()` like `BraveTorClientUpdater::Cleanup()`. WDYT?

Comment (darkdh) [reply browser/extensions/brave_extension_management.cc]: ok, sounds good but you forgot to remove this header https://github.com/brave/brave-core/commit/81bae87c04cc0572256df7fdb34e26155cfd757a#diff-f7642300415fc6963490faca0750da8bL10

Comment (simonhong) [reply browser/extensions/brave_extension_management.cc]: @darkdh Done.

Comment (bridiver) [reply browser/extensions/brave_extension_management.cc]: @simonhong can you also double-check that deps are removed/added for this file and the file it was added to?

Comment (simonhong) [reply browser/extensions/brave_extension_management.cc]: I checked whole files that touched in this PR by using below commands and added more missing deps.
`gn check out/Debug //brave/browser/extensions | grep brave_extension_management.cc`
`gn check out/Debug //brave/browser/extensions | brave_tor_client_updater.cc`
`gn check out/Debug //brave/test::brave_browser_tests | grep brave_policy_browsertest.cc`
`gn check out/Debug //brave/browser/tor | grep tor_profile_esrvice.cc`
`gn check out/Debug //brave/browser/ui | grep brave_browser_command_controller.cc`
`gn check out/Debug //brave/browser/ui | grep brave_app_menu_model.cc`
`gn check out/Debug //brave/test:brave_browser_tests | grep  brave_app_menu_model_browsertest.cc'
`gn check out/Debug //brave/browser/ui | grep brave_profile_menu_view_helper.cc`

**Comment by simonhong** [comment]
CI seems not success but some fails are not related with this PR(audit and ios failure).
ios failure occured at the early stage maybe some internal error and it was also success in previous ci build.

**Comment by tomlowenthal** [comment]
This should cover brave/brave-browser#6584 too.
